test/libyul/yulOptimizerTests/unusedStoreEliminator/mcopy_different_areas_full_words.yul

```diff
@@ -0,0 +1,30 @@
+{
+    mstore(0x40, 42)
+    mstore(0x60, 42)
+    mstore(0x80, 123)
+    mstore(0xa0, 123)
+    // FIXME: The second mcopy is not redundant but disappears too
+    mcopy(0, 0x40, 0x40)
+    mcopy(0, 0x80, 0x40)
```

Comment:
Looks like there's something wrong either with my implementation or with the `UnusedStoreEliminator`. I have already found several cases where `mcopy()` gets removed even when not redundant.

Reply:
I was about to say that so far there haven't been instructions that simultaneously read and write to/from potentially even the same memory region - but actually calls can do that as well (reading arguments from, and writing returndata to memory within the same instruction...) - however calls can never be removed, so they're fine in that sense at least, so it may still be that that's not accounted for properly.

Reply:
If it significantly simplifies things, we can just exclude ``mcopy`` as a candidate for removal for now - but we still need to make sure that its effects are properly accounted for with respect to the other reads and writes as a minimum.

Reply:
It'd be good to have at least some clarity on what exactly makes removing it problematic (if the write of ``mcopy`` is unused, the reading part should also not matter, so I'm not exactly sure without digging into it, what the problem actually is)

Reply:
In the end this turned out to be my fault. I defined semantic info for `MCOPY` as consisting of two separate `SemanticInformation::Operation`s, one for read, one for write. Turns out this is the first one of this kind that `UnusedStoreEliminator` has to deal with. It had an assert against it and I relaxed it but I overlooked the spot where it stores the operation. As a result it stored only the read and ignored the write.

But yeah, due to the fact that it's prepared to store only one operation per statement, it would require additional changes to handle removing `MCOPY`. So I'm going with your suggestion to exclude `MCOPY`. It looks like with that the reads over writes are properly accounted for because when it goes over blocks, it can handle multiple operations per statement just fine. I'm now adding test coverage to make sure it works in all situations, and there's quite a few cases, but still fewer than if I had to vover the removal of `MCOPY` as well.